RTL Support Addition

superav opened the pull request:
Layout files have been edited and changed so that they allow support for lanugauge scripts that read from right to left and those that read left to right.

superav committed:
Feat: Addition of RTL Support

These changes allow support of language scripts that read both right to
left and left to right.

tarun0 commented on app/src/main/res/layout/activity_loan_account_details.xml:
```diff
@@ -15,6 +15,7 @@
         android:gravity="center_vertical"
         android:orientation="horizontal"
         android:paddingLeft="@dimen/row_padding_left"
```
This line is redundant as the minApiVersion is 17. `android:paddingStart="@dimen/row_padding_left"` in itself is sufficient.

tarun0 reviewed (changes requested):
Missed [this](https://github.com/superav/self-service-app/blob/dbfc32e81774e89e5065f5b59400478dd64cfdb6/app/src/main/res/layout/activity_saving_account_details.xml#L35) maybe?

Please make these small changes and submit the task again.

superav committed:
Feat: Addition of RTL Support

These changes allow support of language scripts that read both right to
left and left to right. I have removed the redundant line as per the
comment to the previous pull request.

superav committed:
Feat: Addition of RTL Support

These changes allow support of language scripts that read both right to
left and left to right. I have fixed the line as per the comment in the
previous pull request.

tarun0 commented:
@superav Please squash the 3 commits into a single commit. Then @therajanmaurya will review it finally before merging.

superav committed:
These changes allow support of language scripts that read both right to
left and left to right.

superav committed:
Merge branch 'RTL_Support_Addition' of https://github.com/superav/self-service-app into RTL_Support_Addition

superav committed:
Revert "These changes allow support of language scripts that read both right to"

This reverts commit 8a4a695802eb05ec3cf48e39d8e7be59a92fc674.

superav committed:
Feat: Addition of RTL Support

These changes allow support of language scripts that read both right to
left and left to right.

mdew192837 reviewed (commented):
@tarun0 I have a few questions. Just trying to learn. :) Should we leave paddingLeft for backwards compatibility? You mentioned miniApiVersion 17. Does that mean paddingLeft is not needed? Also, should we add android:supportsRtl="true" in the AndroidManifest.xml file?

tarun0 commented:
@mdew192837 Yes, to support RTL, one must enable it in the manifest file by using `android:supportsRtl="true"`. Since the app's minSdkVersion is 17, it means that this app can't be installed in the Android versions below Jellybean (any API < 17). The attributes such as paddingLeft are used with API<17 (or if RTL is disabled in manifest). 
Since RTL is already enabled (this is why you can't find the change in the commits) and the minSdkVersion for the app is 17, we don't need marginLeft because marginLeft and marginStart will mean the same.

Hope that clarifies your query.
